packages/firebase_analytics/android/src/main/java/io/flutter/plugins/firebaseanalytics/FirebaseAnalyticsPlugin.java

```diff
@@ -16,20 +22,55 @@
 import java.util.Map;
 
 /** Flutter plugin for Firebase Analytics. */
-public class FirebaseAnalyticsPlugin implements MethodCallHandler {
-  private final PluginRegistry.Registrar registrar;
-  private final FirebaseAnalytics firebaseAnalytics;
+public class FirebaseAnalyticsPlugin implements MethodCallHandler, FlutterPlugin, ActivityAware {
+  private FirebaseAnalytics firebaseAnalytics;
+  private MethodChannel methodChannel;
+  private Activity activity;
 
   public static void registerWith(PluginRegistry.Registrar registrar) {
-    final MethodChannel channel =
-        new MethodChannel(registrar.messenger(), "plugins.flutter.io/firebase_analytics");
-    channel.setMethodCallHandler(new FirebaseAnalyticsPlugin(registrar));
+    FirebaseAnalyticsPlugin instance = new FirebaseAnalyticsPlugin();
+    instance.setActivity(registrar.activity());
+    instance.onAttachedToEngine(registrar.context(), registrar.messenger());
   }
 
-  private FirebaseAnalyticsPlugin(PluginRegistry.Registrar registrar) {
-    this.registrar = registrar;
-    FirebaseApp.initializeApp(registrar.context());
-    this.firebaseAnalytics = FirebaseAnalytics.getInstance(registrar.context());
+  @Override
+  public void onAttachedToEngine(@NonNull FlutterPluginBinding binding) {
+    onAttachedToEngine(
+        binding.getApplicationContext(), binding.getFlutterEngine().getDartExecutor());
+  }
+
+  private void onAttachedToEngine(Context applicationContext, BinaryMessenger binaryMessenger) {
+    FirebaseApp.initializeApp(applicationContext);
+    firebaseAnalytics = FirebaseAnalytics.getInstance(applicationContext);
+    methodChannel = new MethodChannel(binaryMessenger, "plugins.flutter.io/firebase_analytics");
+    methodChannel.setMethodCallHandler(this);
+  }
+
+  @Override
+  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
+    firebaseAnalytics = null;
+    setActivity(null);
```

Comment:
I'm not sure that this line will ever have an effect. Activity should always be null by the time this runs.

Reply:
Good to know, would it also already be null for the other `detach` callbacks like `onDetachedFromActivityForConfigChanges` etc?

Reply:
Removed activity reset from "Detach" callbacks.